types/File.ts

```diff
@@ -16,8 +16,8 @@ export interface FormattingOptions extends FormattingOverrides {
 // to customize the output
 // Be careful with indentation if the output syntax is indentation-sensitive (e.g. python, yaml)
 export interface FormattingOverrides {
-  commentStyle?: 'short' | 'long' | 'none';
-  commentPosition?: 'above' | 'inline';
+  commentStyle?: 'short' | 'long' | 'none'; // refers to JS object in lib/enums/commentStyles.js
```

Comment:
@jorenbroekema Here I am still struggeling with how we could use the `transformTypes` enum for replacing these hardcoded string values for the different Transforms:
```typescript
// types/Transform.ts
export type NameTransform = BaseTransform<'name', string>;
export type AttributeTransform = BaseTransform<'attribute', Record<string, unknown>>;
export type ValueTransform = BaseTransform<'value', unknown | undefined>;
```

Reply:
Further, do you think there is anything for the `types/Format.ts` that we could an enum for, in order to be more precise with the typings? Not sure here, because custom formats and custom transforms always have to be possible and I am therefore uncertain if it makes any sense to type it something like:
```typescript
import { formats } from '../lib/enums/index.js';
type formats = typeof formats;

export interface Format {
  name: string | formats[keyof formats];
  format: FormatFn;
}
```

Reply:
And one more thing, which I noticed in [the docs of the format-helpers](https://github.com/amzn/style-dictionary/blob/main/docs/src/content/docs/reference/Utils/format-helpers.md). There are existing different `options.formatting` string values, which we have not covered yet as an enum?

Reply:
```ts
import { transformTypes as transformTypesEnum } from '../lib/enums/transformTypes.js';

export type NameTransform = BaseTransform<typeof transformTypes.name, string>;
export type AttributeTransform = BaseTransform<typeof transformTypes.attribute, Record<string, unknown>>;
export type ValueTransform = BaseTransform<typeof transformTypes.value, unknown | undefined>;
```

Assuming the transformTypes have been strictly typed with the typecasts I described, if you then check e.g.:
```ts
type foo = NameTransform['type']; // typed as "name"
```